src/Geometry/conversions.jl

```diff
@@ -25,6 +25,29 @@ LocalVector(
     local_geometry::LocalGeometry{I},
 ) where {T, I} = local_geometry.∂ξ∂x' * u
 
+# conversions between Covariant and Contravariant vectors
+Contravariant123Vector(
+    u::CovariantVector{T, (1, 2)},
+    local_geometry::LocalGeometry{(1, 2, 3)},
+) where {T} = local_geometry.gⁱʲ * Covariant123Vector(u[1], u[2], zero(u[1]))
+
+Contravariant123Vector(
+    u::CovariantVector{T, (3,)},
+    local_geometry::LocalGeometry{(1, 2, 3)},
+) where {T} =
+    local_geometry.gⁱʲ * Covariant123Vector(zero(u[1]), zero(u[1]), u[1])
+
+
+ContravariantVector(
+    u::CovariantVector{T, I},
+    local_geometry::LocalGeometry{I},
+) where {T, I} = local_geometry.gⁱʲ * u
+
+CovariantVector(
+    u::ContravariantVector{T, I},
+    local_geometry::LocalGeometry{I},
+) where {T, I} = local_geometry.gᵢⱼ * u
```

Comment:
Is diagonalization along `I` required here?

Reply:
(just out of curiosity)

Reply:
This works the full vector, with all `ndim` components. The first two deals with partial vectors!

Reply:
Ok, I think you're saying yes. The generic method already defined does not diagonalize:

```julia
CovariantVector(
    u::ContravariantVector,
    local_geometry::LocalGeometry{I},
) where {I} = project(
    CovariantAxis{I}(),
    project(ContravariantAxis{I}(), u),
    local_geometry,
)
```